src/javascripts/components/copy.mjs

```diff
@@ -19,34 +20,50 @@ class Copy {
 
     this.$button = document.createElement('button')
     this.$button.className = 'app-copy-button'
-    this.$button.setAttribute('aria-live', 'assertive')
     this.$button.textContent = 'Copy code'
 
+    this.$status = document.createElement('span')
+    this.$status.className = 'govuk-visually-hidden'
+    this.$status.setAttribute('aria-live', 'assertive')
+
+    this.$module.insertAdjacentElement('beforebegin', this.$status)
     this.$module.insertAdjacentElement('beforebegin', this.$button)
-    this.copyAction()
+
+    const $clipboard = new ClipboardJS(this.$button, {
+      target: (trigger) => trigger.nextElementSibling
+    })
+
+    $clipboard.on('success', (event) => this.successAction(event))
+    $clipboard.on('error', (event) => this.resetAction(event))
   }
 
   /**
-   * Set up button copy action
+  * Copy to clipboard success
+  *
+  * @param {import('clipboard').Event} event
+  */
```

Comment:
ESLint says this one's gone a little bit wonky and needs a description

They're only warnings so we don't fail on them in this repo
https://github.com/alphagov/govuk-design-system/actions/runs/6339165139/job/17217788790

```suggestion
   * Copy to clipboard success
   *
   * @param {import('clipboard').Event} event - Clipboard event
   */
```

Reply:
Hmm, I thought we had Prettier fix this stuff up on commit, or am I thinking of the frontend repo? (Or something else entirely? I'm on a new laptop so don't have any extensions running yet!)